### ISSUE TITLE
"Block is already registered" errors when having other plugins with custom blocks

### ISSUE THREAD

**Issue by htmgarcia** [publishpress/PublishPress-Blocks#58]
### Steps to reproduce

1. Install and activate WooCommerce or other plugin with custom blocks
2. Go to PP Blocks > Profiles > "Default"
3. Disable a random block and save
4. By having browser console opened, edit a page or post
5. A list of errors below is outputted in console (only one time after step 4):

```
Reloading editor by PublishPress Blocks plugin
Block "core/paragraph" is already registered.
Block "core/image" is already registered.
Block "core/heading" is already registered.
```

This is the line that triggers the errors: https://github.com/publishpress/PublishPress-Blocks/blob/master/src/assets/js/editor.jsx#L90
Basically is an editor settings override. We tell the editor which blocks to load based on the Profile (step 2 & 3).
_____
Reported in a few old WP topics: 
https://wordpress.org/support/topic/advanced-custom-fields-plugin-2/
https://wordpress.org/support/topic/block-core-is-already-registered/

**Comment by htmgarcia** [comment]
One more report: https://wordpress.org/support/topic/block-core-is-already-registered-2/

**Comment by htmgarcia** [comment]
Status: partially solved.

**Comment by htmgarcia** [comment]
The last report: https://wordpress.org/support/topic/block-core-paragraph-is-already-registered/

**Comment by htmgarcia** [comment]
This is an irregular issue to reproduce. Now only possible with Kadence blocks plugin activated. Maybe their advice could help to figure out this issue as well 201#issuecomment-764144231

**Comment by andergmartins** [comment]
@htmgarcia I was able to reproduce the error.
I was checking and I would like to suggest changing the following line:

https://github.com/publishpress/PublishPress-Blocks/blob/master/src/assets/js/editor.jsx#L90

Maybe instead of using the method `initializeEditor` we could try using `reinitializeEditor`.

Considering the error relates to trying to register all the blocks again, reinitializing makes more sense then another call to the initialize method:

```
wp.editPost.reinitializeEditor(advgb_blocks_vars.post_type, advgb_blocks_vars.post_id, target, new_settings, window._wpGutenbergDefaultPost);
```

**Comment by htmgarcia** [comment]
When a PP Blocks profile is edited or a new plugin with blocks is added, this is the expected result in console. Luckily we reduced to not be outputted often. 

Let's wait on Gutenberg future releases to try to solve 100% this minor issue.

**Comment by htmgarcia** [comment]
This fix seems to get rid of this old bug https://github.com/publishpress/PublishPress-Blocks/commit/64e2dc8166b8f3228a0f2a50981422ec4bf11438#diff-a987bdca7d43b1546618778d13c24beb7f9e352cac509083909da9accb9c1b22R29323-R29331
